lib/mayaUsd/ufe/UsdUndoAttributesCommands.h

```diff
@@ -110,6 +110,35 @@ class UsdRemoveAttributeCommand : public UsdUndoableCommand<Ufe::UndoableCommand
     const std::string _name;
 }; // UsdRemoveAttributeCommand
 
+#ifdef UFE_V3_FEATURES_AVAILABLE
+class UsdSetMetadataCommand : public UsdUndoableCommand<Ufe::UndoableCommand>
+{
+public:
+    UsdSetMetadataCommand(
+        MayaUsd::ufe::UsdAttribute& attr,
+        const std::string&          key,
+        const Ufe::Value&           newValue);
+    ~UsdSetMetadataCommand() override;
+
+    // Delete the copy/move constructors assignment operators.
+    UsdSetMetadataCommand(const UsdSetMetadataCommand&) = delete;
+    UsdSetMetadataCommand& operator=(const UsdSetMetadataCommand&) = delete;
+    UsdSetMetadataCommand(UsdSetMetadataCommand&&) = delete;
+    UsdSetMetadataCommand& operator=(UsdSetMetadataCommand&&) = delete;
+
+    //! Create a UsdSetMetadataCommand
+    static UsdSetMetadataCommand::Ptr
+    create(MayaUsd::ufe::UsdAttribute& attr, const std::string& key, const Ufe::Value& newValue);
+
+    void executeUndoBlock() override;
+
+private:
+    MayaUsd::ufe::UsdAttribute& _attr;
```

Comment:
It feels very awkward to keep a reference to the MayaUsd::ufe::UsdAttribute in the command, as this is usually unsafe, because of questions about the lifescope of the reference.  I realize that it isn't used after executeUndoBlock(), but can you add a comment about this here in the header?  I would even consider storing a plain C pointer to the MayaUsd::ufe::UsdAttribute, and nulling it out after executeUndoBlock() is done, for complete clarity.